engine/function/src/templates/Basic.ftl

```diff
@@ -786,6 +786,63 @@ public class Basic {
         return ifelseObj(new BooleanVectorDirect(condition), new ObjectVectorDirect<T>(trueCase), new ObjectVectorDirect<T>(falseCase));
     }
 
+    /**
+     * Returns elements from either trueCase or falseCase, depending on condition.
+     *
+     * @param condition a boolean value used to select output values.
+     * @param trueCase value returned when condition is true.
+     * @param falseCase value returned when condition is false.
+     * @return An array of T whose values are determined by the corresponding elements of condition, trueCase, and falseCase.
+     *         The result element will be trueCase if the condition element is true;
+     *         falseCase if the condition element is false; or null if the condition element is null.
+     *         Returns null if any of the inputs is null.
+     */
+    public static <T> T[] ifelseObj(BooleanVector condition, T trueCase, T falseCase) {
+        if (condition == null) {
+            return null;
+        }
+
+        final int n_c = condition.intSize("condition");
+        @SuppressWarnings("unchecked") final T[] result;
+
+        if ( trueCase == null && falseCase == null ) {
```

Comment:
If you feel like changing it (different style, but equivalent). I think it's a tad shorter and more readable.

However if you *don't* feel like it, please remove the extra spacing inside the parens for lines 808, 810, 812

```
final T typeToUse = trueCase != null ? trueCase : falseCase;
if (typeToUse == null) {
   throw new IllegalArgumentException("trueCase and falseCase are null.  Can not resolve a return type.");
}
if (trueCase != null && falseCase != null && trueCase.getClass() != falseCase.getClass()) {
    throw new IllegalArgumentException("Inputs have different types. trueCase=" + trueCase.getClass() + " falseCase=" + falseCase.getClass());
}
result = (T[])Array.newInstance(typeToUse.getClass(), n_c);
```